File: Flutter & Dart/Projects/my_litters/backend/src/services/vehicleService.js
const prisma = require('../config/db');

exports.listForUser = (userId) =>
  prisma.vehicle.findMany({
    where: { userId },
    orderBy: { createdAt: 'desc' },
  });

exports.create = async (userId, { plateNumber, make, model, year, fuelType }) => {
  if (!plateNumber || !fuelType) {
    const e = new Error('plateNumber and fuelType are required');
    e.status = 400;
    throw e;
  }
  return prisma.vehicle.create({
    data: { userId, plateNumber, make, model, year, fuelType },
  });
};

exports.removeForUser = async (userId, vehicleId) => {
  const v = await prisma.vehicle.findUnique({ where: { id: vehicleId } });
  if (!v || v.userId !== userId) {
    const e = new Error('Vehicle not found');
    e.status = 404;
    throw e;
  }
  await prisma.vehicle.delete({ where: { id: vehicleId } });
  return { id: vehicleId };
};

File: Flutter & Dart/Projects/my_litters/lib/services/vehicle_service.dart
import '../models/vehicle.dart';
import 'api_client.dart';

class VehicleService {
  VehicleService([ApiClient? client]) : _api = client ?? ApiClient();
  final ApiClient _api;

  Future<List<Vehicle>> list() async {
    final data = await _api.get('/api/vehicles') as List<dynamic>;
    return data
        .map((j) => Vehicle.fromJson((j as Map).cast<String, dynamic>()))
        .toList();
  }

  Future<Vehicle> create({
    required String plateNumber,
    String? make,
    String? model,
    int? year,
    required String fuelType,
  }) async {
    final data = await _api.post('/api/vehicles', body: {
      'plateNumber': plateNumber,
      if (make != null && make.isNotEmpty) 'make': make,
      if (model != null && model.isNotEmpty) 'model': model,
      if (year != null) 'year': year,
      'fuelType': fuelType,
    });
    return Vehicle.fromJson((data as Map).cast<String, dynamic>());
  }

  Future<void> remove(String vehicleId) async {
    await _api.delete('/api/vehicles/$vehicleId');
  }
}
